robot/client/client.go

```diff
@@ -687,16 +696,15 @@ func (rc *RobotClient) Refresh(ctx context.Context) (err error) {
 
 func (rc *RobotClient) updateResources(ctx context.Context) error {
 	// call metadata service.
+
 	names, rpcAPIs, err := rc.resources(ctx)
-	// only return if it is not unimplemented - means a bigger error came up
 	if err != nil && status.Code(err) != codes.Unimplemented {
```

Comment:
I wonder if this should just be `if err != nil` now. If `ResourceNames` returns `Unimplemented`, should that be surfaced?

Reply:
I now feel the `status.Code(err) != codes.Unimplemented` piece of this predicate should be removed. Curious if anyone else disagrees?

Reply:
I think you're right. Since we don't return an err where the `rpcSubtypesUnimplemented` flag is set, it will always be nil when it is `codes.Unimplemented`

Reply:
so after dealing with the merge conflicts, I needed to add this check to not break the `TestClientDialerOption` test.

Reply:
I added it back in since it doesn't directly affect things this PR is for